File: 04StringsV2/Java/StringV2_04.java

public class StringV2_04 {
    public static void main(String[] args) {
        String songLyrics = "You say goodbye, and I say hello.";

        // Using contains() equivalent in Java.
        boolean result = songLyrics.contains("goodbye");
        System.out.println(String.format("%b", result));
        System.out.println(String.format("%b", songLyrics.contains("greetings")));

        // Using startsWith()
        boolean begin1 = songLyrics.startsWith("You");
        System.out.println(String.format("\"%s\" starts with \"%s\" : %b", songLyrics, "You", begin1));
        
        boolean begin2 = songLyrics.startsWith("hello");
        System.out.println(String.format("\"%s\" starts with \"%s\" : %b", songLyrics, "hello", begin2));

        // Using endsWith()
        boolean end1 = songLyrics.endsWith("hello.");
        System.out.println(String.format("\"%s\" ends with \"%s\" : %b", songLyrics, "hello.", end1));
        
        boolean end2 = songLyrics.endsWith("say");
        System.out.println(String.format("\"%s\" ends with \"%s\" : %b", songLyrics, "say", end2));
    }
}

File: 04StringsV2/python/StringV2_04.py
def main():
    songLyrics = "You say goodbye, and I say hello."

    # Using the "in" operator for .Contains() equivalent
    result = "goodbye" in songLyrics
    print(f"{result}")
    print(f"{'greetings' in songLyrics}")

    # Using startswith()
    begin1 = songLyrics.startswith("You")
    print(f"\"{songLyrics}\" starts with \"You\" : {begin1}")

    begin2 = songLyrics.startswith("hello")
    print(f"\"{songLyrics}\" starts with \"hello\" : {begin2}")

    # Using endswith()
    end1 = songLyrics.endswith("hello.")
    print(f"\"{songLyrics}\" ends with \"hello.\" : {end1}")

    end2 = songLyrics.endswith("say")
    print(f"\"{songLyrics}\" ends with \"say\" : {end2}")

if __name__ == "__main__":
    main()
